### DISCUSSION TITLE
Asset Sources

### DISCUSSION THREAD

**Discussion by AndrewRChapman** [stride3d/stride#2583] [General]
Asset Procurement

I have been creating Model Packs for both Unity and Unreal for some years now, hosting on Unreal, Unity and other Game Engine Stores, including the more generic FAB. I was wondering where the users of Stride obtain their assets from. Without a Store available, do you search the internet?

I am looking at potentially creating packs for Stride, for sale on FAB, which will include specific Materials for the Stride Engine.

Is this something the Stride Community/Users feel is worth doing at this time? I envisage creating a zip file, that contains folders of the various components for your scene, that can be simply drag and drop. Whilst the packs would be for Single License/Seat, with a EULA that means you cannot resell any packs of contents within directly, they can be used to incorporate in finished games/projects and fully encapsulated to prevent exposure of the contents.

Is that something the Stride Community/Users see as useful? Feedback encouraged.

**Comment by Doprez** [comment]
FAB would be what I use today, although I have used Turbo Squid and Sketchfab in the past.

I moved away from using premade assets myself as I usually found that I had to change the components back in my Unity days and it became more time consuming. I do see the value in premade models though as long as they fit the style I am going for or can give me a good white boxing set just to get my base idea down, thats something I find Synty is solid for doing as I dont like their aesthetic but they have a lot of base props for different themes.

All that to say I would be more interested in just models myself but I come from a background of being a programmer first so others may have a different view.

**Comment by IXLLEGACYIXL** [comment]
either you have assets to sell like models materials etc which you can sell on other platforms as well
or you have code and there you can just sell your nuget or class library on other platforms as well

so there was never a need for an asset store as there arent restrictions like in other engines that need workarounds for assets

**Comment by AndrewRChapman** [comment]
Selling on multiple platforms is always ideal. It is only really the formats like materials that really needs to be different.

I'm thinking either a dedicated Page on say FAB to sell packs for Stride, or just hosting packs on FAB under FBX/OBJ meshes, and PNG Textures with a host of maps to suit almost every Engine..... more generic but simpler. The only thing I would do is to load a sample set of models into stride, which I have already done anyway, to see it they can be textured okay and what maps are needed.

I cannot just simply add those packs into my current packs on FAB, as there is no Stride Engine listed, so they would have to be Generic with no materials set up.

Although not the full list of asset packs I sell, but this FAB listing should give an indication.
https://www.fab.com/sellers/Immersive-Games
or Unity...
https://assetstore.unity.com/publishers/39958

Of course, the Unity offerings are all FBX/OBJ with PNG texture maps, unlike the Unreal packs, which are formatted for Unreal only. Unity does require that they be downloaded into an Editor, before you can copy/paste the meshes and textures into a folder of your choosing. However that can add a step or two to the process, so I am reaching out to users, to see how they buy stuff like meshes/texture maps etc, and where.

**Reply by IXLLEGACYIXL** [reply]
stride can import fbx or pretty mcuh anything via drag and drop , why do you want them to be converted to a stride asset? 

there is 2 things:
1. materials dont get setup imported, like a blender material shader only works partially, you need to setup it again inside stride sometimes to reattach the maps
2. you cant attach scripts, which makes sense

**Reply by AndrewRChapman** [reply]
I was assuming packs might contain Materials out of the box, like other Editors. There is no generic method of supplying that, so each engine has its own. I was considering seeing if Stride Users wanted that, but having looked closely at Stride, there doesn't seem to be any way of exporting models, textures and compatible materials into a pack anyway.

I think at this time, I might just host generic packs of my wares, in a zipped format with just FBX/Obj and PNG textures. I need to see if the market place is there before spending lots of time on it.

I only attach scripts to objects, as they are Engine specific. Only 2 out of the hundred or so packs I sell, contain scripts. Those are the Indoor and Exterior Security Camera sets.

**Reply by IXLLEGACYIXL** [reply]
> I was assuming packs might contain Materials out of the box, like other Editors. There is no generic method of supplying that, so each engine has its own. I was considering seeing if Stride Users wanted that, but having looked closely at Stride, there doesn't seem to be any way of exporting models, textures and compatible materials into a pack anyway.
> 
> I think at this time, I might just host generic packs of my wares, in a zipped format with just FBX/Obj and PNG textures. I need to see if the market place is there before spending lots of time on it.
> 
> I only attach scripts to objects, as they are Engine specific. Only 2 out of the hundred or so packs I sell, contain scripts. Those are the Indoor and Exterior Security Camera sets.

strides materials are like your blender shader but the importer doesnt 1:1 translate it, most of it is lost

what you need for a material are the maps/textures etc what you can set in the editor which you will do as developer yourself anyway 
so being able to buy the barebones fbxses and the images etc and a readme or whatever would make most sense i think

but thats just my humble opinion

**Reply by NicusorN5** [reply]
If you want to support multiple platforms, I think you're better off with raw assets.

**Reply by AndrewRChapman** [reply]
> > I was assuming packs might contain Materials out of the box, like other Editors. There is no generic method of supplying that, so each engine has its own. I was considering seeing if Stride Users wanted that, but having looked closely at Stride, there doesn't seem to be any way of exporting models, textures and compatible materials into a pack anyway.
> > I think at this time, I might just host generic packs of my wares, in a zipped format with just FBX/Obj and PNG textures. I need to see if the market place is there before spending lots of time on it.
> > I only attach scripts to objects, as they are Engine specific. Only 2 out of the hundred or so packs I sell, contain scripts. Those are the Indoor and Exterior Security Camera sets.
> 
> strides materials are like your blender shader but the importer doesnt 1:1 translate it, most of it is lost
> 
> what you need for a material are the maps/textures etc what you can set in the editor which you will do as developer yourself anyway so being able to buy the barebones fbxses and the images etc and a readme or whatever would make most sense i think
> 
> but thats just my humble opinion

I have used stride many times, seeing how it is progressing, but it doesn't seem to have a built in process where packs containing its own native materials, can be included with meshes and textures, apart from a manual process outside the editor itself. That was my goal, to see how portable the materials are from an engine. The meshes and textures are already on my system in a central depository for use in a raft of engines/editors. That could be linked to having no fixed destination like a FAB or AssetStore destination. Hence why I am here, looking for feedback to the process currently in use.

I use many different programmes to create my meshes/texture maps, no materials are used in the export, I create each one within the platform editor I am selling for. That way I can control how my goods look like. Even if that only represents a starting point for Scene Developers. It feels sort of naked, just passing meshes and textures, if that is all that can be done.

**Reply by Eideren** [reply]
You just have to archive your assets .sd* files and materials and the model and texture files, if the path in the .sd* files are relative, users just have to drop their sd files in their asset folder and the fbx in resource at the same path you have them at and it should work.
You could also create a project specifically to host your files, users would just have to extract it and reference the project in their game solution. Although that one is a bit more involved on your side, and might pollute the users solution UI.

**Reply by AndrewRChapman** [reply]
Hi,
Thank you for the feedback. I have taken the route of providing a 'raw' package that includes the texture maps and meshes. I think this will be more productive and less reliant on specific materials/shaders or Editor versions.

I have listed the first package for free, just to get a lay of the land.

https://www.fab.com/listings/3a19868a-68d4-47cb-b5a0-580161377c54

Feedback would be appreciated on how you find the process. I know scaling can be an issue with different platforms (metric vs imperial), as well as the Normal formats. Where possible, it is more controllable, if the 'strength' of the normal can be a negative as well as a positive value to switch between them.

There is both FBX and OBJ versions on the download page.

**Comment by TKTSWalker** [comment]
Not sure who to speak to but if Stride is looking to have an asset store made, I'm more than happy to help development! You can see a software store I am developing below;

https://www.artstation.com/artwork/nJ8YXe

I'm between jobhunting and finishing projects though so I can't go %1000 on it just yet; still, i'd like to start if anything!

**Reply by w0wca7a** [reply]
If you mean that Stride needs services to publish assets, then the [roadmap ](https://github.com/orgs/stride3d/projects/13) does not currently contain this.

**Reply by TKTSWalker** [reply]
I meant that I would be willing to develop them! I think that having a place where people could look at assets (and possibly monetize them) could be a great incentive to get more developers on Stride

**Reply by w0wca7a** [reply]
I have already done a similar project. on Word Press and ecommerce with online viewing of models. And also a Windows application for importing resources into the project.
Problems that I encountered:
1. Manual file export, Game Studio needs for additional functionality.
2. Import. The same as export. I downloaded the asset as FBX and used Win-importer. 6 mb FBX model after extract has raised to 120 mb folder. + manual download, + manual import.

Commercial services will probably be difficult to implement, since it will require registration of a trade mark or something else for financial transactions.
